### DISCUSSION TITLE
lolMiner 1.45

### DISCUSSION THREAD

**Discussion by Lolliedieb** [Lolliedieb/lolMiner-releases#1264] [General]
# lolMiner 1.45
(hold on while we prepare the Windows builds)

### Changes
- Added Ethash + Alephium dual mining mode analogous to existing Ethash + Ton mode. Use --dualmode ALEPHDUAL to use it. Supported GPUs: Nvidia Turing & Ampere generation, AMD Polaris (RX 400, 500), Navi and Big Navi generations.
- Stratum connections will now have an increasing timeout (steps of 5 seconds) when reconnecting plus a small random time of 1 second in order to not overload the target pool in case of recovering from a network outtake. 
- lolMiner API now has new page /gui, which allows you to watch your rig mining in a web browser. Navigate to http://<ip of your rig>:<apiport>/gui in your brower doing so. 
- Added experimental (at the moment a bit slow) Ethash kernels for new AMD BC-250 APUs
- In dual mining the algorithm that mined a share will now be shown in command line output
- --ton-mode 6 stratum mode will now obey the prefix send to it on the first bytes of the nonce (request from other pools that want to use this scheme)
- Startup time of dual mining on non-LHR cards is now reduced to 30 seconds after DAG build
- Removed --ton-mode 2 from automatic detection (you still can request it explicitly) - ton-pool.com will instead use mode 1 for existing and mode 6 for upcoming new stratum servers. 

### Fixes
- Fixed multiple stratum bugs - mostly crash fixes when running web socket based connections

Preview of lolMiner web monitoring:
![image](https://user-images.githubusercontent.com/40234439/154066036-4bd454f4-e41c-4909-a5f3-8ba0f5447d18.png)

<hr /><em>This discussion was created from the release <a href='https://github.com/Lolliedieb/lolMiner-releases/releases/tag/1.45'>lolMiner 1.45</a>.</em>

**Comment by GuRuNa** [comment]
Great web UI. 👍👍👍

**Reply by rashrrr** [reply]
dualdevices dont work for ETH+ALPH dual mining

**Reply by vincentmaximo** [reply]
ETH+ALPH not working in hiveos because version 1.45 is not already implemented

**Reply by plutochang** [reply]
dualdevices dont work for ETH+ALPH dual mining in version 1.45.
device 0 (1050ti): ALPH only
device 1 (3090): ETH + ALPH

Parameter:
--dualmode ALEPHDUAL
--dualpool hk.alephium.herominers.com:1199
--dualworker %WORKER_NAME%dual
--dualuser {ALPH_WALLET}
--dualdevices 0
--maxdualimpact \*,*

**Reply by tirtaleonardi** [reply]
I would like to address the same problem, ETH + ALPH doesn't seem to work. To add, solo algorithm mining ETH also didn't work when I tried.

When I downgraded to ver 1.44, the algorithm initiated normally.

@Lolliedieb

**Reply by dawidmosk** [reply]
still not working alphdual, any solutions? lolMiner crashing

**Comment by nirotub** [comment]
Added RTX 3080 12GB support?

**Reply by Lolliedieb** [reply]
Well that we had in 1.44 already, although the unlock is rather poor. Nv did worsen the lhr game again.

**Reply by nirotub** [reply]
> 

On version 1.44, the miner could not select the LHR parameters for 3080 12GB

**Reply by nirotub** [reply]
> Well that we had in 1.44 already, although the unlock is rather poor. Nv did worsen the lhr game again.

Are you trying to unlock 3080 12GB?

**Comment by Adikul1** [comment]
how to use api in hiveos?

**Comment by YaakovLi** [comment]
So how to log in the mine pool GUI can you explain in detail

**Reply by YaakovLi** [reply]
and how to find the apiport and the rig port

**Reply by Adikul1** [reply]
type this in bat file --apihost ip --apihost 8080

**Reply by YaakovLi** [reply]
thx,i will try now

**Reply by YaakovLi** [reply]
> 在 bat 文件中输入 --apihost ip --apihost 8080

so i can define the ‘ip’ by myself?Are there any limitations to its definition

**Reply by Adikul1** [reply]
no, modem gives every device one ip. Google how to find out ip for your computer or youtube

**Reply by YaakovLi** [reply]
> no, modem gives every device one ip. Google how to find out ip for your computer or youtube

thx，but  I wanna know whether the 'IP' you mentioned refers to the public IP of the mining machine or the LAN IP. My computer does not have a public IP

**Comment by Adikul1** [comment]
no lan ip is required

⁣Get BlueMail for Android ​

On Feb 15, 2022, 10:25 PM, at 10:25 PM, YaakovLi ***@***.***> wrote:
>> no, modem gives every device one ip. Google how to find out ip for
>your computer or youtube
>
>thx，but  I wanna know whether the 'IP' you mentioned refers to the
>public IP of the mining machine or the LAN IP. My computer does not
>have a public IP
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/Lolliedieb/lolMiner-releases/discussions/1264#discussioncomment-2182481
>You are receiving this because you commented.
>
>Message ID:
***@***.***>

**Comment by BitMaci** [comment]
Nice job, I am dual mining on Windows 10 ETC+TON (no LHR) and it flows great. Keep the good job going!

**Reply by sammey946** [reply]
hi  i start  min  eth + ton lost conecting to next ton pool have fix issus pls!!!

**Comment by rashrrr** [comment]
dualdevices dont work for ETH+ALPH dual mining

**Comment by Fummelum** [comment]
Is there a way to combine mining rigs GUI, so I can watch all my rigs in one GUI?
Right now I need to have a new tab for every rig.

**Comment by r230** [comment]
Hello, problem to run on this pool, on version 1.44 all works good.Thx
Pool https://next.ton-pool.com timeout or send defect message: "<unspecified file>(1): expected value"

**Reply by ZeNch25** [reply]
same here with same pool (inclusive next.ton-pool.club)
1.43/1.44 works fine.

**Reply by Ivanbox888** [reply]
Me to catch this error

**Reply by brianmonk** [reply]
try to add `--ton-mode 2`
or use --pool wss://eu1.stratum.ton-pool.com/stratum
worked for me

**Reply by Les982** [reply]
This  --ton-mode 2 worked for me now on 1.45 no errors thank you

**Reply by sammey946** [reply]
write here  code total

**Comment by SaturnV8** [comment]
Please add fan control function option, just simple fixed fan speed eg --fan 80,75,60,100
Thank you!

**Comment by thiti2202** [comment]
Can't connect to ton-pool server, but version 1.44 can use normally.

**Reply by nerzon** [reply]
Connected to: https://next.ton-pool.com:443 (84ms latency) 
connection and mining hashrate is ok, but in telegram reported 0 hashrate

**Reply by nerzon** [reply]
TON-Pool News, [15.02.2022 23:26]
[Переслано из TON-Pool News]
Let us clarify the situation here.
lolMiner 1.45 has new features in ton-mode 6 that are required to work with our new endpoints, and thanks to @Lolliedieb for that. However, it also dropped support for ton-mode 2. When running it against the original endpoint https://next.ton-pool.com, it sends a query every second, and we are not prepared to handle so many requests from our miners. Therefore, we do not allow using lolMiner 1.45 with https://next.ton-pool.com, and it will stop working in the future.
The new endpoint is not well-tested and we cannot guarantee that it will work flawlessly. However, for miners who want to try it out, please use wss://eu1.stratum.ton-pool.com/stratum as the argument to --pool option.
Right now, we recommend staying at lolMiner 1.43 before we get the new endpoint tested out. Once we get the new endpoint working, there may be more endpoints to come. Thanks for your support!

**Comment by hlviktor** [comment]
Hello, I have a problem. I have 2x AMD RX6600. My controller crashes when I want to use this tool and work with it. Temperature is stable from GPU and CPU. The GPU is in normal settings.

Crash occurs irregularly.

Another tool PhoenixMiner works more stable (with the same settings)

**Comment by lei32bryansk** [comment]
Greetings, gentlemen.
 
 I sincerely thank you for the work done, you are really great.
  However, I encountered a problem with low performance on the Palit geforce rtx 3080 ti GPU n Hive OS, as a result of which it issues 70 Mh / s on ETH and approximately 1.5 Gh / s on TON, which is much lower than its real capabilities. 
 I kindly ask you to fix it asap. 

Thanks a lot for attention, best regards.

**Comment by ralva1982** [comment]
6 часов вокеры майнили в никуда. 0 хэша за 6 часов, а на 1,44 все отлично, Я зол на вас(((

**Comment by Les982** [comment]
I'm currently mining ETH and TON on RaveOS looking into this new version 1.45 were ETH and ALPH can be mined is it possible to mine all three?

**Comment by maxxcd** [comment]
In HiveOS, if I want to add proxy parameters, what should I fill in?

-- proxy * * *. * * *. * * *. * * * : 7893?

**Comment by hendra2222** [comment]
Can someone help me what would be the reason for not getting TON hashrate when mining dual ETH and TON? I have 4 cards:
3060 LHR
3060 TI LHR
2070super 
6700

2070 and 6700 I am getting TON hashrate of 780 and 816 and eth 45mh/s and 38.9mh/s
but both my LHR Cards are getting 0 TON hashrate .. This happens since release of ETH and TON support

my 3060LHR = gigabyte 3060 12gb ram
3060 ti = Gigabyte 3060 ti 8gb ram

This is not the case when mining TON Coin only. I am getting all together 5000+ Mh/s...

**Reply by hendra2222** [reply]
This works now. nvmd.

**Comment by aminlv** [comment]
1.45 seems doesnt work with https://next.ton-pool.com
i keep getting errors can someone post correct method plz

**Reply by Les982** [reply]
--ton-mode 2

Fixed it for me but I stopped using 1.45 because unlike 1.43 I didn't get constant hashrate. 1.45 still needs work for sure

**Reply by aminlv** [reply]
tnX fixed my issue2 but as u said seems not properly working with dual mining
well back to prev version for now

**Comment by aminlv** [comment]
good news for ppl having RX 580 series version 1.45 work dual mining ETH+TON w.o any issues as far as i tested

have good farming

**Comment by MaikerS55** [comment]
How to completely disable LHR auto-search? I have a REV 1 RTX 3060 and I don't need lhr auto lookup, it only reduces the hashrate!

**Comment by dotstal** [comment]
Hello, tell me how to remove unnecessary output in the win10 miner window: New job received....

**Comment by AndreJoker** [comment]
постоянно отваливается хешрейт тон в дуале тон+етх и алефио в дуале алефио+етх или падает в 2 раза, риг из 1070 и 3090, если фармить только тон все норм

**Reply by dotstal** [reply]
Раздели отельно майнер на каждую карту, мне помогло, теперь стабильно, к том уже найдёшь ту, которая валит весь риг.

**Reply by AndreJoker** [reply]
как в батнике прописать для винды подскажи

**Reply by dotstal** [reply]
--devices 0
